Samples/StorageProviders/StorageProviders/BaseJSONStorageProvider.cs

```diff
@@ -77,7 +86,7 @@ public Task Close()
         public async Task ReadStateAsync(string grainType, GrainReference grainReference, IGrainState grainState)
         {
             if (DataManager == null) throw new ArgumentException("DataManager property not initialized");
-            var entityData = await DataManager.Read(grainState.GetType().Name, grainReference.ToKeyString());
+            var entityData = await DataManager.Read(grainState.State.GetType().Name, grainReference.ToKeyString());
```

Comment:
Sounds reasonable. The state type was what the previous version was using and i didn't change it for these updates. All of these changes I made will produce the same results as before.